src/server/pps/server/api_server.go

```diff
@@ -362,14 +362,14 @@ const (
 
 // authorizePipelineOp checks if the user indicated by 'ctx' is authorized
 // to perform 'operation' on the pipeline in 'info'
-func (a *apiServer) authorizePipelineOp(ctx context.Context, operation pipelineOperation, input *pps.Input, output string) error {
+func (a *apiServer) authorizePipelineOp(ctx context.Context, operation pipelineOperation, input *pps.Input, projectName, outputName string) error {
 	return a.txnEnv.WithReadContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
-		return a.authorizePipelineOpInTransaction(txnCtx, operation, input, output)
+		return a.authorizePipelineOpInTransaction(txnCtx, operation, input, projectName, outputName)
 	})
 }
 
 // authorizePipelineOpInTransaction is identical to authorizePipelineOp, but runs in the provided transaction
-func (a *apiServer) authorizePipelineOpInTransaction(txnCtx *txncontext.TransactionContext, operation pipelineOperation, input *pps.Input, output string) error {
+func (a *apiServer) authorizePipelineOpInTransaction(txnCtx *txncontext.TransactionContext, operation pipelineOperation, input *pps.Input, projectName, outputName string) error {
```

Comment:
it doesn't look like we're using the `projectName` param here

Reply:
Not used yet — I thought of this as part of making PPS project-aware, even when auth itself currently doesn’t care about projects.

Does raise the question, though: if passing the project name, and passing the output repo name, and the output repo name is always the pipeline name, then should this be another instance of passing a `*pps.Pipeline`? Or since there are actual repo checks, should it be passing `*pfs.Repo` instead?

Reply:
good callout. pfs.Repo would make more sense